src/security/ErrorDetail.cc

```diff
@@ -716,18 +724,20 @@ Security::ErrorDetail::err_lib_error() const
 size_t
 Security::ErrorDetail::convert(const char *code, const char **value) const
 {
+    // TODO change this to return SBuf
     typedef const char *(ErrorDetail::*PartDescriber)() const;
     static const std::map<const char*, PartDescriber> PartDescriberByCode = {
         {"ssl_subject", &ErrorDetail::subject},
         {"ssl_ca_name", &ErrorDetail::ca_name},
         {"ssl_cn", &ErrorDetail::cn},
         {"ssl_notbefore", &ErrorDetail::notbefore},
         {"ssl_notafter", &ErrorDetail::notafter},
-        {"err_name", &ErrorDetail::err_code},
+        {"err_name", &ErrorDetail::err_code_legacy},
         {"ssl_error_descr", &ErrorDetail::err_descr},
         {"ssl_lib_error", &ErrorDetail::err_lib_error}
     };
 
+    // TODO why are we doing this? can't we just lookup the map above?
```

Comment:
I bet we are looping because the original code (before 83b053a) was looping and changing from a loop to a map lookup is not trivial. The original code had to loop because it used a raw/POD array.